src/suggestion-box.ts

```diff
@@ -0,0 +1,383 @@
+                } else {
+                    this.setInputValue();
+                }
+                break;
+        }
+    }
+    InputEvent(e: any) {
+        this.prevIndex = 0;
+        this.index = -1;
+        this.input_el = e.target;
+        if (this.list) {
+            if ((this.input_el as HTMLInputElement)?.value) {
+                this.filtered_list = this.list.filter((item) =>
+                    item[this.listkey].includes(
+                        (this.input_el as HTMLInputElement)?.value
+                    )
+                );
+
+                this.setSuggestion(this.filtered_list, false);
+            } else {
+                this.filtered_list = [];
+
+                this.setSuggestion(this.filtered_list, false);
+            }
+        }
+        return;
+    }
+
+    setActive(resultLength: number) {
+        if (this.prevIndex >= 0 && this.prevIndex < resultLength) {
+            const prevItem = this.private_items[this.prevIndex];
+            if (prevItem) {
+                prevItem.classList.remove("sgn-active");
+            }
+        }
+        if (this.index >= 0 && this.index < resultLength) {
+            const selectedItem = this.private_items[this.index];
+
+            if (selectedItem) {
+                if (this.prevIndex != -1) {
+                    this.private_items[this.prevIndex].classList.remove(
+                        "sgn-active"
+                    );
+                }
+                selectedItem.classList.add("sgn-active");
+            }
+        } else if (this.index == resultLength && this.has_more) {
+            this.private_items[this.index].classList.add("sgn-active");
+        }
+
+        this.prevIndex = this.index;
+    }
+
+    setInputValue() {
+        const selectedValue = this.result[this.index]?.name;
+        this.prevIndex = this.index;
+        if (selectedValue) {
+            (this.input_el as HTMLInputElement).value = selectedValue;
+        }
+    }
+    _highlight(name: any) {
+        const input = (this.input_el as HTMLInputElement)?.value;
+        if (input && name.toLowerCase().includes(input.toLowerCase())) {
+            const parts = name.split(new RegExp(`(${input})`, "gi"));
+
+            return parts
+                .map((part: any, index: number) => {
+                    if (index !== 0) {
+                        return `<span class="sgn-highlight">${part}</span>`;
+                    } else {
+                        return part;
+                    }
+                })
+                .join("");
+        }
+        return name;
+    }
+    _startsWithHighlight(name: any) {
+        const input = (this.input_el as HTMLInputElement)?.value;
+        if (input && name.toLowerCase().startsWith(input.toLowerCase())) {
+            const start = name.substring(0, input.length);
+            const rest = name.substring(input.length);
+            return `<span class="sgn-highlight" >${start}</span>${rest}`;
```

Comment:
This entire block can be used as 

`input_val = input_val.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');`
`re = new RegExp("(" + input_val.split(' ').join('|') + ")", "gi");`
`return name.replace(re, "<span class="sgn-highlight">$1</span>")`

The idea is to highlight all the occurences of the matched text.